lib/tasks/reporting.rake

```diff
@@ -73,6 +75,43 @@ namespace :reporting do
     end
   end
 
+  desc "A CSV report of all documents published by the given organisation"
+  task organisation_documents: :environment do
+    options = opts_from_environment(:organisation_slug)
+
+    CSV.open("#{options[:organisation_slug]}-documents.csv", 'w') do |csv|
+      csv << ['Content ID', 'Path', 'Title', 'Format', 'First Published', 'Last Updated']
+
+      # So we have the organisation association for all edition types
+      Edition.include(Edition::Organisations)
+
+      org_id = Organisation.where(slug: options[:organisation_slug]).limit(1).pluck(:id).first
+
+      scope = Edition.
+        joins(:edition_organisations, :document, :translations).
+        includes(:document, :translations).
+        where(
+          edition_organisations: { lead: 1, organisation_id: org_id },
+          state: %w{published}
+        )
+
+      progress_bar = ProgressBar.create(format: "%e [%b>%i] [%c/%C]", total: scope.count)
+
+      scope.find_each do |edition|
+        first_published = edition.document.ever_published_editions.order(:id).limit(1).pluck(:public_timestamp).first
```

Comment:
I'm finding it a little tricky visualising the data coming back and wondering if we can get duplicate data?
i.e. can `scope` have multiple rows (essentially `editions`) per `document`? in which case is it possible that we end up with duplicate rows in the final csv?

Reply:
Good catch, I missed that between the version I used and the version I committed. Added a `distinct(:document_id)`.